### ISSUE TITLE
Test fails with Compared using equal?, which compares object identity, but expected and actual are not the same object.

### ISSUE THREAD

**Issue by pravi** [licensee/licensee#503]
### Describe the bug

```
Failures:

  1) Licensee confidence threshold exposes the inverse of the confidence threshold
     Failure/Error: expect(described_class.inverse_confidence_threshold).to be(0.02)

       expected #<Float:22430> => 0.02
            got #<Float:22440> => 0.02

       Compared using equal?, which compares object identity,
       but expected and actual are not the same object. Use
       `expect(actual).to eq(expected)` if you don't care about
       object identity in this example.
     # ./spec/licensee_spec.rb:46:in `block (3 levels) in <top (required)>'

  2) Licensee::ContentHelper knows the similarity
     Failure/Error: expect(subject.similarity(subject)).to be(100.0)

       expected #<Float:22450> => 100.0
            got #<Float:22460> => 100.0

       Compared using equal?, which compares object identity,
       but expected and actual are not the same object. Use
       `expect(actual).to eq(expected)` if you don't care about
       object identity in this example.
     # ./spec/licensee/content_helper_spec.rb:65:in `block (2 levels) in <top (required)>'

  3) Licensee::Matchers::Dice returns the match confidence
     Failure/Error: expect(subject.confidence).to be(100.0)

       expected #<Float:22470> => 100.0
            got #<Float:22480> => 100.0

       Compared using equal?, which compares object identity,
       but expected and actual are not the same object. Use
       `expect(actual).to eq(expected)` if you don't care about
       object identity in this example.
     # ./spec/licensee/matchers/dice_matcher_spec.rb:30:in `block (2 levels) in <top (required)>'

Finished in 12.32 seconds (files took 1.06 seconds to load)
2388 examples, 3 failures, 18 pending

Failed examples:

rspec ./spec/licensee_spec.rb:45 # Licensee confidence threshold exposes the inverse of the confidence threshold
rspec ./spec/licensee/content_helper_spec.rb:63 # Licensee::ContentHelper knows the similarity
rspec ./spec/licensee/matchers/dice_matcher_spec.rb:29 # Licensee::Matchers::Dice returns the match confidence

Randomized with seed 8414
```
### Steps to reproduce the behavior

Run test suite on i386 architecture.

### Expected behavior

Comparing float-pointing numbers for equality is always a recipe for failure

### Additional context
Full build log
https://ci.debian.net/data/autopkgtest/testing/i386/r/ruby-licensee/15051694/log.gz

**Comment by mlinksva** [comment]
Hi @pravi! Good to know licensee mostly works on i386. Supporting that probably isn't a priority for the project, but it's an easy fix in #504. Thanks for reporting this, and for maintaining the ruby-licensee package in Debian!

**Comment by pravi** [comment]
@mlinksva thanks for fixing it quickly.
